include/odp/api/spec/init.h

```diff
@@ -150,17 +157,29 @@ typedef struct odp_init_t {
 	    worker and control masks do not overlap.
 	 */
 	const odp_cpumask_t *control_cpus;
+
 	/** Replacement for the default log fn */
 	odp_log_func_t log_fn;
+
 	/** Replacement for the default abort fn */
 	odp_abort_func_t abort_fn;
+
 	/** Unused features. These are hints to the ODP implementation that
 	 * the application will not use any APIs associated with these
 	 * features. Implementations may use this information to provide
 	 * optimized behavior. Results are undefined if applications assert
 	 * that a feature will not be used and it is used anyway.
 	 */
 	odp_feature_t not_used;
+
+	/** Shared memory parameters */
+	struct {
+		/** Maximum memory usage in bytes. This is the maximum
+		 *  amount of shared memory that application will reserve
+		 *  concurrently. */
+		uint64_t max_memory;
```

Comment:
0 - means not specified. And what about continuous of memory chunks? Requesting  one big continues shared memory chunk is not general solution.

Reply:
@muvarov `odp_shm_capability()` already tells the application the largest contiguous size it can reserve (`max_size`) and the maximum number of reserves it can do (`max_blocks`). This is just hinting to the implementation the total size of all reserves the application will do.

Reply:
I'll add comment about. Although, I already changed documentation to require param_init() call and say that don't change values that you are not going to use (init sets it to zero).